axum/src/routing/mod.rs

```diff
@@ -512,6 +519,19 @@ where
     }
 }
 
+fn replace_trailing_slash(uri: &Uri, new_path: &str) -> Uri {
+    let mut new_path_and_query = new_path.to_string();
+    if let Some(query) = uri.query() {
+        new_path_and_query.push('?');
+        new_path_and_query.push_str(query);
+    }
+
+    let mut parts = uri.clone().into_parts();
+    parts.path_and_query = new_path_and_query.parse().ok();
```

Comment:
Feels to me like this should be an `unwrap`

```suggestion
    parts.path_and_query = new_path_and_query.parse().unwrap();
```